components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java

```diff
@@ -1188,4 +1197,37 @@ protected void unsetRealmService(RealmService realmService) {
 
         OAuth2ServiceComponentHolder.getInstance().setRealmService(null);
     }
+
+    @Reference(
+            name = "claim.resolver",
+            service = ClaimResolver.class,
+            cardinality = ReferenceCardinality.MULTIPLE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetClaimResolver"
+    )
+    protected void setClaimResolver(ClaimResolver claimResolver) {
+
+        ClaimResolver existingClaimResolver =
+                OAuth2ServiceComponentHolder.getInstance().getClaimResolver();
+
+        if (existingClaimResolver != null) {
+            log.warn("Multiple claim resolvers are registered. Claim resolver:"
```

Comment:
there is a claim provider registered by default in the activate method, so doen't this trigger a warn log always when there is an external provider registered

Reply:
It will trigger a warn when there is an external provider registered. Should I remove the warn log?